UXToolsGame/Plugins/UXTools/Source/UXTools/Private/Interactions/UxtGenericManipulatorComponent.cpp

```diff
@@ -105,7 +112,11 @@ bool UUxtGenericManipulatorComponent::GetOneHandRotation(const FTransform& InSou
 			FQuat OrientationSwing, OrientationTwist;
 			Orientation.ToSwingTwist(FVector::UpVector, OrientationSwing, OrientationTwist);
 
-			OutTargetTransform.SetRotation(Orientation);
+			// Previously the orientation was simply set back to the same value
+			//OutTargetTransform.SetRotation(Orientation);
+
+			//Let's use the Twist component we isolated above to set orientatin such that it always has an upvector matching world.up.
+			OutTargetTransform.SetRotation(OrientationTwist);
```

Comment:
good catch! - can you remove the previous code that was using the wrong rotation? we don't want to keep outdated code as comments around :)